test/Table_Tests/src/Common_Table_Operations/Add_Group_Number_Spec.enso

```diff
@@ -0,0 +1,131 @@
+
+            g0 = t.add_group_number (..Unique on=['x', 'y']) "g" from=10 step=3
+            g0.at 'g' . to_vector . should_equal [10, 13, 16, 13, 10, 16]
+
+        group_builder.specify "should add group number by equal counts, with from and step" <|
+            t = table_builder [['x', [1, 2, 3, 4, 5]], ['y', [5, 4, 3, 2, 1]], ['z', [1, 5, 4, 2, 3]]]
+
+            g0 = t.add_group_number (..Equal_Count 3) "g" from=10 step=3
+            g0.at 'g' . to_vector . should_equal [10, 10, 13, 13, 16]
+
+        group_builder.specify "must specify group_by with Unique" <|
+            t = table_builder_from_rows ['x', 'y', 'z'] [[1, 0, 2], [0, 1, 0], [1, 2, 0], [0, 1, 1], [1, 0, 1], [1, 2, 1]]
+
+            t.add_group_number ..Unique "g" . should_fail_with Missing_Argument
+
+        group_builder.specify "should report floating point equality warning when grouping on float columns" <|
+            t = table_builder_from_rows ['x', 'y', 'z'] [[1.0, 0.0, 2.0], [0.0, 1.0, 0.0], [1.0, 2.0, 0.0], [0.0, 1.0, 1.0], [1.0, 0.0, 1.0], [1.0, 2.0, 1.0]]
+            g0 = t.add_group_number (..Unique on=['x', 'y']) "g"
+            Problems.expect_warning Floating_Point_Equality g0
+
+        group_builder.specify "should fail if columns provided in group_by do not exist" <|
+            t = table_builder_from_rows ['x', 'y', 'z'] [[1, 0, 2], [0, 1, 0], [1, 2, 0], [0, 1, 1], [1, 0, 1], [1, 2, 1]]
+
+            t.add_group_number (..Unique on=['x', 'q']) . should_fail_with Missing_Input_Columns
+
+        group_builder.specify "should fail if columns provided in order_by do not exist" <|
+            t = table_builder [['x', [1, 2, 3, 4, 5]], ['y', [5, 4, 3, 2, 1]], ['z', [1, 5, 4, 2, 3]]]
+
+            t.add_group_number (..Equal_Count 3 order_by=['q']) "g" . should_fail_with Missing_Input_Columns
+
+        group_builder.specify "will fail if the row number exceeds Long range" <|
+            max_long = Java_Long.MAX_VALUE
+
+            t = table_builder_from_rows ['x', 'y', 'z'] [[1, 0, 2], [0, 1, 0], [1, 2, 0], [0, 1, 1], [1, 0, 1], [1, 2, 1]]
+
+            Problems.assume_no_problems <| t.add_group_number (..Unique on=['x', 'y']) "g" from=(max_long - 10)
+
+            t2 = t.add_group_number (..Unique on=['x', 'y']) "g" from=(max_long - 1)
+            t2.should_fail_with Illegal_Argument
+            t2.catch.to_display_text . should_contain "The row number has exceeded the 64-bit integer range"
+
+        group_builder.specify "should rename existing column upon a name clash, and attach a warning" <|
+            t = table_builder_from_rows ['x', 'y', 'z'] [[1, 0, 2], [0, 1, 0], [1, 2, 0], [0, 1, 1], [1, 0, 1], [1, 2, 1]]
+
+            g0 = t.add_group_number (..Unique on=['x', 'y']) "y"
+            g0.at 'y 1' . to_vector . should_equal [0, 1, 2, 1, 0, 2]
+            Problems.expect_warning Duplicate_Output_Column_Names g0
```

Comment:
If the values in columns were something else the numbers 0, 1, 2 (e.g. a, b, c) it would be far easier to tell what this test does - as I wasn't really sure if it was checking if `y 1` is the new column or existing column - both will have values 0, 1, 2 just in maybe different order.
Also it would still be good to check not only `y 1` but also `y`.

Reply:
Done